Source/Core/EllipsoidRhumbLine.js

```diff
@@ -0,0 +1,533 @@
+                                                          start.longitude, start.latitude, end.longitude, end.latitude);
+    }
+
+    /**
+     * Initializes a rhumb line on the ellipsoid connecting the two provided planetodetic points.
+     *
+     * @alias EllipsoidRhumbLine
+     * @constructor
+     *
+     * @param {Cartographic} [start] The initial planetodetic point on the path.
+     * @param {Cartographic} [end] The final planetodetic point on the path.
+     * @param {Ellipsoid} [ellipsoid=Ellipsoid.WGS84] The ellipsoid on which the rhumb line lies.
+     */
+    function EllipsoidRhumbLine(start, end, ellipsoid) {
+        var e = defaultValue(ellipsoid, Ellipsoid.WGS84);
+        this._ellipsoid = e;
+        this._start = new Cartographic();
+        this._end = new Cartographic();
+
+        this._heading = undefined;
+        this._distance = undefined;
+        this._ellipticity = undefined;
+        this._ellipticitySquared = undefined;
+
+        if (defined(start) && defined(end)) {
+            computeProperties(this, start, end, e);
+        }
+    }
+
+    defineProperties(EllipsoidRhumbLine.prototype, {
+        /**
+         * Gets the ellipsoid.
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Ellipsoid}
+         * @readonly
+         */
+        ellipsoid : {
+            get : function() {
+                return this._ellipsoid;
+            }
+        },
+
+        /**
+         * Gets the surface distance between the start and end point
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Number}
+         * @readonly
+         */
+        surfaceDistance : {
+            get : function() {
+                //>>includeStart('debug', pragmas.debug);
+                Check.defined('distance', this._distance);
+                //>>includeEnd('debug');
+
+                return this._distance;
+            }
+        },
+
+        /**
+         * Gets the initial planetodetic point on the path.
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Cartographic}
+         * @readonly
+         */
+        start : {
+            get : function() {
+                return this._start;
+            }
+        },
+
+        /**
+         * Gets the final planetodetic point on the path.
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Cartographic}
+         * @readonly
+         */
+        end : {
+            get : function() {
+                return this._end;
+            }
+        },
+
+        /**
+         * Gets the heading from the start point to the end point.
+         * @memberof EllipsoidRhumbLine.prototype
+         * @type {Number}
+         * @readonly
+         */
+        heading : {
+            get : function() {
+                //>>includeStart('debug', pragmas.debug);
+                Check.defined('distance', this._distance);
```

Comment:
I know this is what `EllipsoidGeodesic` has, but this is a really odd check to me, especially since there is no public property called `distance`.  Would it be better for us to update this and `EllipsoidGeodesic` to just be undefined?  On the other hand, it may never happen in practice so I could be bikeshedding.  I'm okay with just punting, but since `EllipsoidGeodesic` is so old, it's good to revisit things once in a while.

Reply:
I think it was just a way to assure that heading and distance are both computed since the constructor allows using `undefined` as `start` and `end`.

We could simply change this to check whether heading is defined, which will make more sense here.

Reply:
Since  EllipsoidGeodescic has been like this forever, I'm probably bike shedding, let's just leave it as-is and we can worry about it if it ever comes up.